### PR TITLE
Migration of pallets to FRAME v2

### PR DESCRIPTION
**Pull request by tankofzion**
This PR aims at:
* Migrate `nft`, `nfts`, `bridge` and `registry` (previously called `va_registry`) to FRAME v2 pallet attribute-like macros.
* Converge existing `nft` and `nfts` pallets into a single `nft` pallet.
* Use new `proofs` library to verify proofs and build bundled hashes.
* Integrate the new `nft`, `bridge`, and `registry` pallets in `Development` runtime.

### CODE REVIEW

File: libs/proofs/src/lib.rs [hunk 1]
```diff
@@ -24,14 +30,18 @@ impl<Hash> Proof<Hash> {
...
+
+	pub fn len(&self) -> usize {
+		self.sorted_hashes.len()
+	}
 }
 
 pub trait Hasher: Sized {
 	/// Hash type we deal with
-	type Hash: Default + AsRef<[u8]> + From<[u8; 32]> + Copy + PartialEq + PartialOrd + Debug;
+	type Hash: Default + AsRef<[u8]> + Copy + PartialEq + PartialOrd + Debug;
```
[the remark this answers is not in the document]
Comment (vedhavyas) [reply libs/proofs/src/lib.rs:41]: why do we need that impl @mustermeiszer ?

**Review by vedhavyas — APPROVED** [review]

**Comment by mustermeiszer** [comment]
@vedhavyas Not sure, what do you mean? Which impl? The `AsMut<[u8]>`

File: libs/proofs/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mustermeiszer) [reply libs/proofs/src/lib.rs:41]: Ah, now I see. Just for convenience. Don't wanna do another PR for this lib, once, we wanna use this. Although, one might correctly argue, that hashes are typically considered to be constant after creation.

File: libs/commons/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+//! The Centrifugians Tribe <tribe@centrifuge.io>
+//!
+//! ## License
+//! GNU General Public License, Version 3, 29 June 2007 <https://www.gnu.org/licenses/gpl-3.0.html>
+
+// ----------------------------------------------------------------------------
+// Imports and dependencies
+// ----------------------------------------------------------------------------
+
+// Common types and traits definition
```
[the remark this answers is not in the document]
Comment (mustermeiszer) [reply libs/commons/src/lib.rs]: @tankofzion

File: pallets/bridge/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+					Origin::signed(RELAYER_C),
+					recipient.clone(),
+					registry_id,
+					token_id.clone(),
+					dest_chain),
+				DispatchError::Module {
+					index: 0,
+					error: 3,
+					Some("InsufficientBalance")});
+			*/
```
[the remark this answers is not in the document]
Comment (mustermeiszer) [reply pallets/bridge/src/tests.rs:182]: @tankofzion

**Review by mustermeiszer — APPROVED** [review]

**Comment by mustermeiszer** [comment]
And can we run the tests again? Was this canceled automatically?

**Pull request merged**